src/main/java/org/jitsi/srtp/crypto/HmacSha1.java

```diff
@@ -15,50 +15,52 @@
  */
 package org.jitsi.srtp.crypto;
 
-import org.bouncycastle.crypto.digests.*;
-import org.bouncycastle.crypto.macs.*;
-import org.jitsi.utils.logging2.*;
-
-import javax.crypto.*;
 import java.security.*;
+import java.util.*;
+import javax.crypto.Mac;
+import org.bouncycastle.jce.provider.*;
+import org.jitsi.utils.logging2.*;
 
 /**
- * Implements a factory for an HMAC-SHA1 <tt>org.bouncycastle.crypto.Mac</tt>.
+ * Implements a factory for an HMAC-SHA1 {@link Mac}.
  *
  * @author Lyubomir Marinov
  */
 public class HmacSha1
 {
     /**
-     * The <tt>Logger</tt> used by the <tt>Aes</tt> class to print out debug
-     * information.
-     */
-    private static final Logger logger = new LoggerImpl(HmacSha1.class.getName());
-
-    /**
-     * Initializes a new <tt>org.bouncycastle.crypto.Mac</tt> instance which
-     * implements a keyed-hash message authentication code (HMAC) with SHA-1.
+     * Initializes a new {@link Mac} instance which implements a keyed-hash
+     * message authentication code (HMAC) with SHA-1.
      *
-     * @return a new <tt>org.bouncycastle.crypto.Mac</tt> instance which
-     * implements a keyed-hash message authentication code (HMAC) with SHA-1
+     * @param parentLogger the logging context
+     * @return a new {@link Mac} instance which implements a keyed-hash message
+     * authentication code (HMAC) with SHA-1
      */
-    public static org.bouncycastle.crypto.Mac createMac()
+    public static Mac createMac(Logger parentLogger)
     {
-        if (OpenSslWrapperLoader.isLoaded())
+        List<Provider> providers = new ArrayList<>(3);
+        if (JitsiOpenSslProvider.isLoaded())
         {
-            return new OpenSslHmac(OpenSslHmac.SHA1);
+            providers.add(new JitsiOpenSslProvider());
         }
-        else
+        providers.add(Security.getProvider("SunJCE"));
```

Comment:
Sure, I can add that. So far, the provider order for HmacSHA1 is static. Should this be made configurable or benchmarked? If so, the Aes class needs to be refactored (or rewritten, I was choking over that code...).

Reply:
Hm, in my WSL Ubuntu 20 the PKCS#11 provider on JDK8 is about as fast/slow as BouncyCastle.

Reply:
I think either configurable or benchmarked, ideally, yeah -- or both -- probably with the configuration key the provider name rather than the class name as the current code does.

The AES-CTR PKCS11 provider is slow (due to limitations/bugs of the PKCS interface, i.e. that it's reconfiguring the key on every pass) but on my Debian on AWS (and on my Mac) it seemed fast to me.  What underlying PKCS library are you using -- NSS?

Reply:
Yes, definitely by using the provider key and not the class name.  IMO the current configuration option is confusing and I'd switch to just using a system property (e.g. org.jitsi.srtp.crypto-provider=...)?

Ubuntu-default, so yes, NSS.

Reply:
Yeah, either a system property or something based on @bbaldino 's new jitsi-metaconfig.